scvimadz/reference/base/_base_reference.py

```diff
@@ -0,0 +1,150 @@
+import importlib
+import os
+import shutil
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import List, Optional, Type, Union
+
+import anndata
+import pandas as pd
+import rich_dataframe
+from anndata import AnnData
+from scvi.model.base import BaseModelClass
+
+from scvimadz.storage.base import BaseStorage
+
+
+class _Obj_Type(Enum):
+    MODEL = "model"
+    DATASET = "dataset"
+
+
+class _Metadata_File(Enum):
+    MODELS_METADATA_FILE = "models_metadata.csv"
+    DATASETS_METADATA_FILE = "datasets_metadata.csv"
+
+
+class BaseReference(ABC):
+    @property
+    @abstractmethod
+    def reference_name(self) -> str:
+        """The name of this reference."""
+        pass
+
+    @property
+    @abstractmethod
+    def model_store(self) -> Type[BaseStorage]:
+        """The backend store for models."""
+        pass
+
+    @property
+    @abstractmethod
+    def data_store(self) -> Type[BaseStorage]:
+        """The backend store for datasets."""
+        pass
+
+    def _get_reference_prefix(self) -> str:
+        return f"{self.reference_name}_"  # e.g. tabula_sapiens_
+
+    def _get_store_for_object(self, obj_type: _Obj_Type) -> Type[BaseStorage]:
+        if not isinstance(obj_type, _Obj_Type):
+            raise ValueError(f"Unrecognized obj_type: {obj_type}")
+        return self.model_store if obj_type is _Obj_Type.MODEL else self.data_store
+
+    def _get_object_keys(self, obj_type: _Obj_Type) -> List[str]:
+        store = self._get_store_for_object(obj_type)
+        keys = [
+            key
+            for key in store.list_keys()
+            if key.startswith(self._get_reference_prefix())
+        ]
+        return keys
+
+    def _list_objects(
+        self,
+        obj_type: _Obj_Type,
+        metadata_fn: _Metadata_File,
+        pretty_print: bool = False,
+    ) -> pd.DataFrame:
+        keys = self._get_object_keys(obj_type)
+        metadata_file = self._get_store_for_object(obj_type).download_file(
+            metadata_fn.value
+        )
+        df = pd.read_csv(metadata_file, index_col="key").loc[keys]
+        if pretty_print:
+            rich_dataframe.prettify(
+                df,
+                row_limit=len(df),
+                col_limit=len(df.columns),
+                clear_console=False,
+            )
+        return df
+
+    def list_models(self, pretty_print: bool = False) -> pd.DataFrame:
```

Comment:
```suggestion
    def get_models_df(self, pretty_print: bool = False) -> pd.DataFrame:
```
list to me seems like it would return a list

Reply:
updated